pvlib/pvsystem.py

```diff
@@ -2274,22 +2303,30 @@ def adrinverter(v_dc, p_dc, inverter, vtol=0.10):
     mppt_hi = inverter['MPPTHi']
     mppt_low = inverter['MPPTLow']
 
-    v_lim_upper = np.nanmax([v_max, vdc_max, mppt_hi])*(1+vtol)
-    v_lim_lower = np.nanmax([v_min, mppt_low])*(1-vtol)
-
-    pdc = p_dc/p_nom
-    vdc = v_dc/v_nom
-    poly = np.array([pdc**0, pdc, pdc**2, vdc-1, pdc*(vdc-1),
-                     pdc**2*(vdc-1), 1/vdc-1, pdc*(1./vdc-1),
-                     pdc**2*(1./vdc-1)])
+    v_lim_upper = np.nanmax([v_max, vdc_max, mppt_hi]) * (1 + vtol)
+    v_lim_lower = np.nanmax([v_min, mppt_low]) * (1 - vtol)
+
+    pdc = p_dc / p_nom
+    vdc = v_dc / v_nom
+    with np.errstate(invalid='ignore', divide='ignore'):
+        poly = np.array([pdc**0,  # replace with np.ones_like?
+                         pdc,
+                         pdc**2,
+                         vdc - 1,
+                         pdc * (vdc - 1),
+                         pdc**2 * (vdc - 1),
+                         1. / vdc - 1,  # divide by 0
+                         pdc * (1. / vdc - 1),  # invalid 0./0. --> nan
+                         pdc**2 * (1. / vdc - 1)])  # divide by 0
     p_loss = np.dot(np.array(ce_list), poly)
     ac_power = p_nom * (pdc-p_loss)
-    p_nt = -1*np.absolute(p_nt)
+    p_nt = -1 * np.absolute(p_nt)
 
     ac_power = np.where((v_lim_upper < v_dc) | (v_dc < v_lim_lower),
                         np.nan, ac_power)
-    ac_power = np.where((ac_power < p_nt) | (vdc == 0), p_nt, ac_power)
-    ac_power = np.where(ac_power > pac_max, pac_max, ac_power)
+    ac_power = np.where(vdc == 0, p_nt, ac_power)
+    ac_power = np.maximum(ac_power, p_nt)
```

Comment:
What do you think about using the `out` parameter here?

Reply:
Either way is fine with me. ``maximum`` and ``minimum`` ignore ``nan`` values, and I figured the ``maximum`` function was easier to read that the ``where``/``out`` pattern.

Reply:
On my computer `np.maximum([1,2,3,np.nan], 2)` gives a runtime warning...

Also, `maximum` has an `out` parameter too.

Reply:
Maybe a numpy version difference? I'm using 1.14.0 in my test environment.

Reply:
I appear to be using 1.11.3.